Backpack/src/main/java/net/skyscanner/backpack/text/BpkTextInputLayout.kt

```diff
@@ -0,0 +1,159 @@
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.view.isVisible
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.colorStateList
+import net.skyscanner.backpack.util.unsafeLazy
+import net.skyscanner.backpack.util.use
+
+class BpkTextInputLayout @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = R.attr.bpkTextInputLayoutStyle,
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+  private val labelView by unsafeLazy<BpkText> {
+    findViewById(R.id.bpk_input_label)
+  }
+
+  private val indicatorView by unsafeLazy<BpkText> {
+    findViewById(R.id.bpk_input_indicator)
+  }
+
+  private val iconSize: Int = resources.getDimensionPixelSize(R.dimen.bpk_icon_size_small)
+
+  private val errorTextColor: ColorStateList
+  private val helperTextColor: ColorStateList
+
+  var editText: BpkTextField? = null
+
+  var errorIcon: Drawable? = null
+    set(value) {
+      field = value?.mutate()
+      field?.setBounds(0, 0, iconSize, iconSize)
+      updateErrorIconTint()
+      updateIndicator()
+    }
+
+  var label: String? = null
+    set(value) {
+      field = value
+      labelView.text = value
+      labelView.isVisible = value != null
+    }
+
+  var error: String? = null
+    set(value) {
+      field = value
+      editText?.hasError = value != null
+      updateIndicator()
+    }
+
+  var helperText: String? = null
+    set(value) {
+      field = value
+      updateIndicator()
+    }
+
+  init {
+    orientation = VERTICAL
+    inflate(context, R.layout.view_bpk_text_input_layout, this)
+    context.obtainStyledAttributes(
+      attrs,
+      R.styleable.BpkTextInputLayout,
+      defStyleAttr, 0
+    ).use {
+      errorTextColor = it.getColorStateList(R.styleable.BpkTextInputLayout_textInputErrorTextColor)
+        ?: ColorStateList.valueOf(context.getColor(R.color.bpkPanjin))
+      helperTextColor = it.getColorStateList(R.styleable.BpkTextInputLayout_textInputHelperTextColor)
+        ?: ColorStateList.valueOf(context.getColor(R.color.__textInputLayoutHelperColor))
+      label = it.getString(R.styleable.BpkTextInputLayout_android_label)
+      error = it.getString(R.styleable.BpkTextInputLayout_textInputError)
+      helperText = it.getString(R.styleable.BpkTextInputLayout_textInputHelperText)
+      errorIcon = it.getDrawable(R.styleable.BpkTextInputLayout_textInputErrorIcon)
+        ?: AppCompatResources.getDrawable(context, R.drawable.bpk_information_circle_sm)
+      labelView.setTextColor(
+        colorStateList(
+          color = context.getColor(R.color.bpkTextPrimary),
+          disabledColor = context.getColor(R.color.__textFieldTextDisabled)
+        )
+      )
+      updateErrorIconTint()
+    }
+  }
+
+  override fun childDrawableStateChanged(child: View?) {
+    super.childDrawableStateChanged(child)
+    if (child is FrameLayout) {
```

Comment:
User can add any amount of FrameLayouts to the component. We can replace it with id check.